src/common/mfu_daos.h

```diff
@@ -38,7 +44,8 @@ int daos_setup(
   char** argpaths,
   daos_args_t* da,
   mfu_file_t* mfu_src_file,
-  mfu_file_t* mfu_dst_file
+  mfu_file_t* mfu_dst_file,
+  bool* is_posix_copy
```

Comment:
@dsikich I still keep getting hung up on the `is_posix_copy` variable, mainly because it wouldn't be extensible if other filesystems are introduced.
We don't necessarily need to address this now, but I had another thought...
What if `mfu_file_t` had three types:
```
enum                 {POSIX, DFS, DAOS} type;
```
`DFS` would be what the existing `DAOS` is now (using the DFS API), and `DAOS` would become the object-level copy (using the DAOS API directly).
Of course, we would want to enforce that `DAOS` can only be used with another `DAOS`

Reply:
@daltonbohning this was more along the lines of what I was thinking, so I like it better than the other option you proposed. Using the posix variable for this version version was to help me quickly separate the code that was only used for POSIX copies, but I agree it should fit into the framework we already have.

Reply:
@daltonbohning I do think looking through the code it is going to take a good amount of renaming and changes to do it this way, even if they are simple, so I think it would be better to improve it this way in another PR.

Reply:
@dsikich I'm okay with that, as long as @adammoody agrees.